lib/rules/for-direction.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * @fileoverview enforce "for" loop update clause moving the counter in the right direction.(for-direction)
+ * @author Aladdin-ADD<hh_2013@foxmail.com>
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        docs: {
+            description: "enforce \"for\" loop update clause moving the counter in the right direction.",
+            category: "Possible Errors",
+            recommended: false
+        },
+        fixable: null,
+        schema: []
+    },
+
+    create(context) {
+
+        return {
+            ForStatement(node) {
+
+                /**
+                 * report an error.
+                 * @returns {void}
+                 */
+                function report() {
+                    context.report({
+                        node,
+                        message: "enforce \"for\" loop update clause moving the counter in the right direction."
+                    });
+                }
+
+                /**
+                 * check UpdateExpression add/sub the counter
+                 * @param {ASTNode} update UpdateExpression to check
+                 * @param {string} counter variable name to check
+                 * @returns {int} if add return 1, if sub return -1, if nochange, return 0
+                 */
+                function updateDirection(update, counter) {
+                    if (update.argument.name === counter) {
+                        if (update.operator === "++") {
+                            return 1;
+                        }
+                        if (update.operator === "--") {
+                            return -1;
+                        }
+                    }
+                    return 0;
+                }
+
+                /**
+                 * check AssignmentExpression add/sub the counter
+                 * @param {ASTNode} update AssignmentExpression to check
+                 * @param {string} counter variable name to check
+                 * @returns {int} if add return 1, if sub return -1, if nochange, return 0
+                 */
+                function assignmentDirection(update, counter) {
```

Comment:
Generally we like to have function names start with verbs (to contrast them with variables/objects/properties, which should be nouns). So a better name would be `getAssignmentDirection` (or maybe `getAssignmentExpressionDirection`).

Also, could this function be moved outside of the ForExpression visitor? As it is, this function is created every time a ForStatement is encountered in the code and it looks like a pure function (doesn't depend on the `node` from the ForStatement visitor function params).